Containerfile.base

```diff
@@ -1,4 +1,6 @@
-FROM public.ecr.aws/docker/library/ruby:3.0.2-slim-bullseye AS ruby-upstream
+ARG RUBY_VERSION=3.0.6
```

Comment:
Would the goal later be to use `cat .ruby-version` here?

Reply:
This is already done as part of the builds: see L12 and L42 of the build script included here.

Basically what this is doing is declaring a FROM arg (this arg won't exist in the image itself) that has a default value (so that there's never an invalid image path being referenced and causing cryptic errors), but we're always going to be overriding it over CLI.